File: MultiLanguageSandbox/src/test/deps/ruby/41.rb


# Counts the number of pairs of students who are considered "equally matched opponents".
# Two students are "equally matched" if the difference in scores for each subject (Chinese, Mathematics, English) 
# is no more than 5 points, and the total score difference is no more than 10 points.
# The function takes an array of arrays, where each sub-array represents a student's scores in the three subjects,
# and returns the total count of "equally matched" student pairs.
#
# Example:
# >>> count_matching_pairs([[90, 90, 90], [85, 95, 90], [80, 100, 91]])
# 2


def count_matching_pairs(students_scores)
  count = 0
  n = students_scores.size

  # Iterate through all possible pairs of students
  (0...n).each do |i|
    (i+1...n).each do |j|
      student1 = students_scores[i]
      student2 = students_scores[j]

      # Check each subject's score difference
      chinese_diff = (student1[0] - student2[0]).abs
      math_diff = (student1[1] - student2[1]).abs
      english_diff = (student1[2] - student2[2]).abs

      # Check total score difference
      total_diff = (student1.sum - student2.sum).abs

      # If all conditions are met, increment the count
      if chinese_diff <= 5 && math_diff <= 5 && english_diff <= 5 && total_diff <= 10
        count += 1
      end
    end
  end

  count
end

raise 'Test failed' unless count_matching_pairs([[90, 90, 90], [85, 95, 90], [80, 100, 91]]) == 2
raise 'Test failed' unless count_matching_pairs([[100, 100, 100], [100, 100, 105], [90, 90, 90]]) == 1
raise 'Test failed' unless count_matching_pairs([[90, 90, 90], [85, 80, 85], [80, 85, 80], [76, 81, 85]]) == 2
raise 'Test failed' unless count_matching_pairs([[100, 100, 100]]) == 0
raise 'Test failed' unless count_matching_pairs([]) == 0

File: MultiLanguageSandbox/src/test/deps/php/41.php
<?php

/**
 * Counts the number of evenly matched pairs of students based on their exam scores.
 * Two students are considered evenly matched if the difference in each subject's score
 * is not greater than 5 and the total score difference is not greater than 10.
 * Input: An integer N representing the number of students followed by N lines of three integers each
 *         representing the scores of each student in Chinese, Mathematics, and English respectively.
 * Output: An integer representing the number of evenly matched pairs of students.
 */

function countEvenlyMatchedPairs($N, $students) {

    $count = 0;
    
    // Iterate through all possible pairs of students
    for ($i = 0; $i < $N; $i++) {
        for ($j = $i + 1; $j < $N; $j++) {
            $student1 = $students[$i];
            $student2 = $students[$j];
            
            // Calculate differences in each subject
            $chineseDiff = abs($student1[0] - $student2[0]);
            $mathDiff = abs($student1[1] - $student2[1]);
            $englishDiff = abs($student1[2] - $student2[2]);
            
            // Calculate total score difference
            $totalDiff = abs(array_sum($student1) - array_sum($student2));
            
            // Check if all conditions are met
            if ($chineseDiff <= 5 && $mathDiff <= 5 && $englishDiff <= 5 && $totalDiff <= 10) {
                $count++;
            }
        }
    }
    
    return $count;
}
function main(){
    assert(countEvenlyMatchedPairs(3, [[90, 90, 90], [85, 95, 90], [80, 100, 91]]) === 2);
    assert(countEvenlyMatchedPairs(3, [[100, 100, 100], [85, 95, 90], [80, 100, 91]]) === 1);
    assert(countEvenlyMatchedPairs(4, [[100, 100, 100],[100, 100, 100],[100, 100, 100],[100, 100, 100]]) === 6);
    
}


main();
?>